allennlp/data/dataset_readers/stanford_sentiment_tree_bank.py

```diff
@@ -0,0 +1,136 @@
+                    for subtree in parsed_line.subtrees():
+                        instance = self.text_to_instance(subtree.leaves(), subtree.label())
+                        if instance is not None:
+                            yield instance
+                else:
+                    instance = self.text_to_instance(parsed_line.leaves(), parsed_line.label())
+                    if instance is not None:
+                            yield instance
+
+    @overrides
+    def text_to_instance(self, tokens: List[str], sentiment: str = None) -> Instance:  # type: ignore
+        """
+        We take `pre-tokenized` input here, because we don't have a tokenizer in this class.
+
+        Parameters
+        ----------
+        tokens : ``List[str]``, required.
+            The tokens in a given sentence.
+        sentiment ``str``, optional, (default = None).
+            The sentiment for this sentence.
+
+        Returns
+        -------
+        An ``Instance`` containing the following fields:
+            tokens : ``TextField``
+                The tokens in the sentence or phrase.
+            label : ``LabelField``
+                The sentiment label of the sentence or phrase.
+        """
+        # pylint: disable=arguments-differ
+        text_field = TextField([Token(x) for x in tokens], token_indexers=self._token_indexers)
+        fields: Dict[str, Field] = {"tokens": text_field}
+        if sentiment is not None:
+            # 0 and 1 are negative sentiment, 2 is neutral, and 3 and 4 are positive sentiment
+            # In 5-class, we use labels as is.
+            # 3-class reduces the granularity, and only asks the model to predict
+            # negative, neutral, or positive.
+            # 2-class further reduces the granularity by only asking the model to
+            # predict whether an instance is negative or positive.
+            if self._granularity == "3-class":
+                if int(sentiment) < 2:
+                    sentiment = "0"
+                elif int(sentiment) == 2:
+                    sentiment = "1"
+                else:
+                    sentiment = "2"
+            elif self._granularity == "2-class":
+                if int(sentiment) < 2:
+                    sentiment = "0"
+                elif int(sentiment) == 2:
+                    return None
+                else:
+                    sentiment = "1"
+            fields['label'] = LabelField(sentiment)
+        return Instance(fields)
+
+    @classmethod
+    def from_params(cls, params: Params) -> 'StanfordSentimentTreeBankTokensDatasetReader':
+        token_indexers = TokenIndexer.dict_from_params(params.pop('token_indexers', {}))
+        use_subtrees = params.pop('use_subtrees', False)
+        granularity = params.pop_choice('granularity', ["5-class", "3-class", "2-class"], True)
+        lazy = params.pop('lazy', False)
+        params.assert_empty(cls.__name__)
+        return StanfordSentimentTreeBankTokensDatasetReader(
```

Comment:
One argument per line in multi-line lists like this.  Also, we've typically put the first argument inline, and aligned all other lines to that, but I don't have a strong opinion on that.  This might be getting long enough that you want to put it on the next line...